### PR TITLE
Feature: Support DISTINCT in new planner 

### LINKED ISSUE #5343: Feature: Support `DISTINCT` in new planner
In SQL, we can use `DISTINCT` to deduplicate the result set of a query, and produce a multi-set.

```SQL
SELECT DISTINCT a, b FROM t; -- The output won't contain duplicated (a, b)
```

This is essentially equivalent with `GROUP BY a, b`.

### PR DESCRIPTION
**Pull request by ygf11**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support base `DISTINCT` and `DISTINCT` with group by.
## Changelog

- New Feature

## Related Issues

Fixes #5343

### CODE REVIEW

File: query/src/sql/exec/mod.rs [hunk 1]
```diff
@@ -362,6 +363,12 @@ impl PipelineBuilder {
             })?;
         }
 
+        // Since transform has added or did, making group expressions as column expr is safe.
+        group_expressions = group_expressions
```
Comment (xudong963) [inline query/src/sql/exec/mod.rs:367]: The group expression may have been a column, so there may be duplicate conversions here.

Comment (xudong963) [inline query/src/sql/exec/mod.rs:367]: The group expression may have been a column, so there may be duplicate conversions here.

Comment (ygf11) [reply query/src/sql/exec/mod.rs:367]: Yes, if a group expression is a column, it is unnecessary here.

The desitnation of adding this code is to make distinct with group-by work.
Two ways to make it better:
1. We do this only when aggregation is generated by distinct.
2. Move this when bind distinct.

File: query/src/sql/exec/mod.rs [hunk 2]
```diff
@@ -364,6 +366,12 @@ impl PipelineBuilder {
             })?;
         }
 
+        // Since transform has been added, making group expressions as column expr is safe.
+        group_expressions
+            .iter_mut()
+            .filter(|expr| matches!(expr, Expression::Column(_)).not())
+            .for_each(|expr| *expr = Expression::Column(expr.column_name()));
+
```
Comment (leiysky) [inline query/src/sql/exec/mod.rs:374]: Maybe we can use a `iter().filter().map().collect()` instead?

Comment (ygf11) [reply query/src/sql/exec/mod.rs:374]: If the original group expression is a column, then it will not be in the collect result.

**Review by leiysky — APPROVED** [review]

**Pull request merged**